### PR TITLE
Benji/2636/probing screen

### PR DESCRIPTION
**Pull request by BenjiPestell**
# Added probing screen (similar to homing)

[Jira Ticket](https://yetitoolsmartsoftware.atlassian.net/browse/YTSS-2636?atlOrigin=eyJpIjoiOTlhY2EzYTAyNGRhNGNiMDhlMDUyODU5N2E3MmRiZjQiLCJwIjoiaiJ9)
- [x] Ready for review

## Checklist
- [x] I have completed a self review
- [ ] I have updated the [spreadsheet](https://docs.google.com/spreadsheets/d/1277nPOKtBFOk_kCo870_zA460-qlw_kGnYlLPwtnadE/edit#gid=0)
- [x] I have updated the jira ticket
- [x] I have added the relevant labels to this PR
- [x] I have updated documentation (if applicable)
- [x] I have run the unit tests suite and they pass
- [x] I have updated the [change log](https://docs.google.com/document/d/1t3s6lQuUcug1Fj8E_i2gBJNvXN0-funwBArG2hsia6k/edit)

## Description

Sort of does what it says on the tin really

## Notes

I'm using the clock instead of binding because I was unable to bind to machine state

## Dependencies for merge
- [ ] [#<pull_request_number>]

### Visual Test
- [ ] Not applicable
- [x] Own computer
- [x] Console 7"
- [ ] Console 10"

### Function Test
- [ ] Not applicable
- [x] Own computer
- [x] Console 7"
- [x] Console 10"

### Unit Tests
- [ ] Not applicable
- [x] Completed

## Screenshots (if applicable)

### CODE REVIEW

File: src/asmcnc/core_UI/components/buttons/probe_button.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            self.sm.add_widget(self.probing_screen)
+
+    def update_image_size(self, instance, value):
+        self.image.size = value
+
+    def update_image_pos(self, instance, value):
+        self.image.pos = value
+
+    def open_screen(self, *args):
+        print("Opening probing screen")
```
Comment (archiepeterjarvis) [inline src/asmcnc/core_UI/components/buttons/probe_button.py:52]: remove print statement

File: src/asmcnc/core_UI/components/buttons/probe_button.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        self.sm.current = 'probing'
+
+    def close_screen(self, *args):
+        print("Closing probing screen")
```
Comment (archiepeterjarvis) [inline src/asmcnc/core_UI/components/buttons/probe_button.py:56]: remove print statement

File: src/asmcnc/core_UI/components/buttons/probe_button.py (hunk 1 above)
Comment (archiepeterjarvis) [reply src/asmcnc/core_UI/components/buttons/probe_button.py:52]: (I initially had these to simulate opening/closing, but I don't think it's worth polluting logs further)

File: src/asmcnc/core_UI/components/buttons/probe_button.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    background_color = (0, 0, 0, 0)
+
+    def __init__(self, router_machine, screen_manager, localization):
+        super(ProbeButton, self).__init__()
+
+        self.sm = screen_manager
+        self.m = router_machine
+        self.l = localization
+
+        self.image = Image(source="./asmcnc/skavaUI/img/z_probe.png", size = self.size, pos = self.pos, allow_stretch = True)
```
Comment (archiepeterjarvis) [inline src/asmcnc/core_UI/components/buttons/probe_button.py:29]: Change path to a relative path, I sent doc in #software-general if needed

File: src/asmcnc/skavaUI/screen_probing.py [hunk 4]
```diff
@@ -125,7 +125,18 @@ def on_enter(self):
         if self.m.is_spindle_on():
             # Spindle is on, need to turn it off
             log("Spindle is on, turning off before probing")
-            self.m.turn_off_spindle()
+
+            # If parent screen is 'home', spindle should be turned off using the spindle button
+            if self.parent_button.return_screen == 'home':
```
Comment (Masc999) [inline src/asmcnc/skavaUI/screen_probing.py:130]: I don't like it. The probing screen should not need to know that there might be a blinking spindle button somewhere. It just turns off the spindle in router_machine.
The blinking spindle button should bind to a spindle_on BooleanProperty that is provided by router_machine and turn its blinking on or off accordingly.

**Review by Masc999 — CHANGES_REQUESTED** [review]

**Review by Masc999 — APPROVED** [review]

**Pull request merged**